workflow/artifacts/oss/oss.go

```diff
@@ -148,6 +152,53 @@ func (ossDriver *ArtifactDriver) ListObjects(artifact *wfv1.Artifact) ([]string,
 	return files, err
 }
 
+func (ossDriver *ArtifactDriver) setBucketLifecycleRule(client *oss.Client, artifact *wfv1.Artifact) error {
```

Comment:
suggestion - rather than `Artifact` could the parameter be `OSSLifecyleRule?

Reply:
Great suggestion. We also need the key and bucket name so I changed this to `OSSArtifact` and removed `ossDriver`.